### PR TITLE
Dala 2633 file download doesnt show status bar

### PR DESCRIPTION
**Pull request by jhaerDfine**
# Pull Request \ Dala 2633 file download doesnt show status bar
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] At least 3 consecutive CI runs are successfully executed to ensure that there are no flaky tests.
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed in order
  - [ ] A fresh clone of dataland.com is generated (see Wiki page on "OTC" for details)
  - [ ] The feature branch is deployed to clone with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the clone server
  - [ ] The feature branch is deployed to dev1 with `Reset non-user related Docker Volumes & Re-populate` turned on, and it's verified that the CD run is green  
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [x] The new feature is manually used/tested/observed on dev server

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -4,7 +4,21 @@
   "requires": true,
   "packages": {
     "": {
-      "name": "Dataland"
+      "dependencies": {
```
Comment (SiegfriedSobkowiak) [inline package-lock.json:7]: Wie kommt diese Änderung zustande? Hätte ich an dieser Stelle nicht erwartet.

File: dataland-frontend/tests/component/components/resources/frameworkDataSearch/DocumentLink.cy.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import { minimalKeycloakMock } from "@ct/testUtils/Keycloak";
+
+describe("check that the progress spinner works correctly for the document link component", function (): void {
+  it("Check that there are no icons before and after triggering a download", function (): void {
+    cy.intercept("**/documents/dummyFile**", {
+      statusCode: 200,
+    }).as("downloadComplete");
+    cy.mountWithPlugins(DocumentLink, {
+      keycloak: minimalKeycloakMock({}),
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (SiegfriedSobkowiak) [inline dataland-frontend/tests/component/components/resources/frameworkDataSearch/DocumentLink.cy.ts:11]: Ist mir nicht klar warum diese 4 Kommentar-Blöcke gebraucht werden. Lokal meckert der linter bei mir nicht und auch die anderen Tests haben es auf den ersten Blick nicht.

File: package-lock.json (hunk 1 above)
Comment (jhaerDfine) [reply package-lock.json]: keine bewusste entscheidung, vermutlich  (versehentlich) entstanden durch test/herum probieren. Rückgängig machen hatte locale keine Auswirkungen.

File: dataland-backend/backendOpenApi.json [hunk 3]
```diff
@@ -14302,16 +14302,16 @@
         ],
         "type": "object",
         "properties": {
-          "companyName": {
+          "countryCode": {
```
[the remark this answers is not in the document]
Comment (jhaerDfine) [reply dataland-backend/backendOpenApi.json]: Das passiert häufiger beim generieren der OpenApi.json (zumindestens bei mir). Wieder auf den Stand von main geändert.

File: gitinfo [hunk 4]
```diff
@@ -1 +1 @@
-{"commit":"e9c1af0edd1f40ff29e85eaae6235244a6c5e5da","time":"2024-05-10T14:13:15+02:00","branch":"DALA-3483_introduce_request_status_update_history"}
+{"commit":"d62aa3f1d17d32c109a1ca5b928f9beb7205491c","time":"2024-05-21T14:01:40+02:00","branch":"DALA-2633_file_download_doesnt_show_status_bar"}
```
[the remark this answers is not in the document]
Comment (jhaerDfine) [reply gitinfo]: als leere Datei neu commited

File: dataland-frontend/tests/component/components/resources/frameworkDataSearch/DocumentLink.cy.ts (hunk 2 above)
Comment (jhaerDfine) [reply dataland-frontend/tests/component/components/resources/frameworkDataSearch/DocumentLink.cy.ts:11]: aus irgendeinem Grund failed rebuild docker-image des e2e Test Container dann : https://github.com/d-fine/Dataland/actions/runs/9210991193
und nachm hinzufügen gehts wieder: https://github.com/d-fine/Dataland/actions/runs/9211217533 
Anderes Tests die beim mount props setzten haben auch diesen kommentar-Block (e.g., RequestReviewButton)

Comment (SiegfriedSobkowiak) [reply dataland-frontend/tests/component/components/resources/frameworkDataSearch/DocumentLink.cy.ts:11]: ok das ist die checkcypresscompilation die da kaputt geht, konnte ich lokal reproduzieren

**Pull request merged**